surefire-booter/src/main/java/org/apache/maven/surefire/booter/BooterDeserializer.java

```diff
@@ -58,6 +60,18 @@ public BooterDeserializer( InputStream inputStream )
         properties = SystemPropertyManager.loadProperties( inputStream );
     }
 
+    /**
+     * Describes the current connection channel used by the client in the forked JVM
+     * in order to connect to the plugin process.
+     *
+     * @return connection string (must not be null)
+     */
+    @Nonnull
+    public String getConnectionString()
+    {
+        return properties.getProperty( FORK_NODE_CONNECTION_STRING );
```

Comment:
as we receive the connection string, we should also receive an auth token.

You can use two separate "properties" or you can create a connection string like
tcp://127.0.0.1:PORT/token=$UUID

the client will send the auth information on the wire as HELO message.
the server can drop the connection in case such token is unknown or in case that such token has already been used

Reply:
@jon-bell 
What is your opinion about using the token?
@eolivelli 
What we solve with this? Nobody knows the port, only the surefire jvm via properties file.
We have to invest more time to extend commadns/events then.

Reply:
@Tibor17 It might be attractive to be able to uniquely identify each client stream based on its unique ID. I don't think that it adds much from a security perspective.

Reply:
Who needs to identify the UUDI?
The JVM and the Maven user dow not need it.
We are forking the JVM with the connection string and every connection string is different. So it cannot happen that they mess up. The information with forkNumber is already in the JVM, so we know and identify each JVM for all the history of Surefire.

Reply:
Any program (even a virus)  can connect to all ports on 127.0.0.1 and mess up the maven process.
If you add a minimal auth you will mitigate this security issue

Reply:
In terms of sequential diagram the auth may slow down the tests due to the server has to wait for the incoming token. Other option is to keep the JVM startup, let it send the token on the top in `ForkedBooter` but dont wait and keep going. If bad token is sent, the `ForkStarter` can nnicely shutdown the JVM and stop the `CommandReader`. Worser situation is when no token is send in the authentication.

Reply:
@eolivelli 
Now I have another idea. What about to inline the token in each response going from JVM to the plugin? Currently we have this frame `:maven-surefire-event:next-test:`. Whatabout to add the token into each frame? Later we will have JSON (gson is small and still alive library) and we reduce a lot of code/encoder using th JSON un/marshaller.

Reply:
@eolivelli 
A new security gap might be in Docker when you `EXPOSE 65000` (expose the port 65000) and the surefire server will find this port unused. This might be a problem right? How can we avoid that?

Reply:
@eolivelli 
@jon-bell 
The Docker container typically maps ports for the wildcard address `0.0.0.0` and not the localhost `127.0.0.1`. So can we exclude the situation in my comment https://github.com/apache/maven-surefire/pull/240/#discussion_r382901754 ?

Reply:
@Tibor17 I don’t see much risk to adding the extra HELO step, the overhead will be on the order of several milliseconds - why not just do it?

Reply:
This patch is not about docker.
Let's keep it away now.
I can imaging your intent, but it is out of scope.

Reply:
I would keep JSON awaus by now, you already developed all of the encode/decoder stuff.
We should move forward and deliver this work.
Adding JSON and any other library will complicate this stuff more and more and there is little benefit, as you already did the work (and it is a great work)

Reply:
@eolivelli 
The task with JSON is of course related to another issue. It is not mandatory here. But it can simplify a lot. Please see the list of modules in Surefire. There is one important for our discussion. It is named `surefire-shadefire` because it contains all shaded artifacts and it's SNAPSHOT version cannot be, in principle, used by other modules due to the compiler. We have to use the release version. Therefore we have to discuss the JSON one release version ahead of time.